Update ERC-4361: Move to Last Call

0xthrpw opened the pull request:
I'd like to request to move this to Last Call.  [SIWE](https://login.xyz/)'s specification has been settled for a while, it has been implemented in Typescript, Rust, Elixir, Python, Ruby and Go and is used in [thousands of projects](https://github.com/spruceid/siwe/network/dependents).

SamWilsn reviewed (dismissed):
On line 41/42, the proposal mentions a couple RFCs. We permit links to RFCs; see EIP-1 for details.

On line 120/121, the proposal mentions ISO 8601. This document is not free to access, and so shouldn't be mentioned in an ERC. Instead, you can use RFC 3339 which is free to access.

---

This PR also needs approval from an initial author (@wyc, @obstropolos, @brantlymillegan, @Arachnid, or @awoie).

0xthrpw commented:
Added links for RFC 5234 and RFC 7405.  Also updated mentions of ISO 8601 to instead specify RFC 3339 as suggested.  I am in contact with Brantly Millegan and this PR was made on his behalf.  Should have his approval and sign off this week.

brantlymillegan reviewed (approved):
lgtm

obstropolos reviewed (approved):
Noting here that the change to `RFC 3339` narrows the formats from the original ISO spec, but I understand in terms of practicality / how SIWE is implemented in practice today (with almost all implementations producing something compatible with `RFC 3339`) - means nothing changes here. It tightens the spec and aligns with current standards, so I'm happy to approve here as well. 

Thanks for pushing this along to get it finalized @brantlymillegan

0xthrpw commented:
hi @SamWilsn, i committed your suggested changes and flipped the flag to allow edits by maintainers. (also thanks for fixing the error on eip7930 that was blowing up the ci/html proofer)
